### PR TITLE
Yielded components with html block content error

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,43 @@
 const visit = require('unist-util-visit');
 var u = require('unist-builder');
 
+const TAG_SPLIT = /([^<]+)?(<[^>]+>)*/
+const mightHaveTag = str => str ? str.indexOf('<') < str.indexOf('>') : false;
+
+let recursion = 0;
```
Comment (josemarluedke) [inline index.js:7]: How is this variable used? It seems we never increment it.

File: index.js [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+
+    return acc;
+  }, []);
+
+  if (result.length === 1) {
+    return [str];
+  }
+
+  if (result.length > 1) {
+    result = result.map(subStr => splitTags(subStr));
```
Comment (josemarluedke) [inline index.js:33]: You can simplify this a bit:

```suggestion
    result = result.map(splitTags);
```

File: index.js [hunk 3]
```diff
@@ -1,7 +1,43 @@
...
+let recursion = 0;
+const splitTags = (str) => {
+  if (recursion > 4) return [str];
+  if (!mightHaveTag(str)) return [str];
+
+  let [_, ...result] = str.match(TAG_SPLIT)
+
+  result = (result || []).filter(Boolean).reduce((acc, subStr, i) => {
+    let previous = acc[i - 1];
+    let previousExists = previous !== undefined;
```
Comment (josemarluedke) [inline index.js:16]: Should we use typeof here?

**Review by josemarluedke — COMMENTED** [review]
This is a really nice change. Thank you for working this out.

I have left a few minor inline comments.

File: index.js (hunk 3 above)
Comment (NullVoxPopuli) [reply index.js:16]: works for me!

File: index.js (hunk 1 above)
Comment (NullVoxPopuli) [reply index.js:7]: whoops, this was a debugging trick I used to get me out of infinite recursion lockups while I was figuring out my exit conditions

**Comment by NullVoxPopuli** [comment]
I'll feedback addressed. thanks!

**Review by josemarluedke — APPROVED** [review]

**Comment by josemarluedke** [comment]
Published under v0.4.0 and in Docfy v0.4.2

**Comment by NullVoxPopuli** [comment]
Thank you!!!! :D

**Pull request merged**